include/libcyphal/transport/can/msg_rx_session.hpp

```diff
@@ -119,6 +119,11 @@ class MessageRxSession final : IRxSessionDelegate, public IMessageRxSession  //
         return std::exchange(last_rx_transfer_, cetl::nullopt);
     }
 
+    void setOnReceiveCallback(OnReceiveCallback::Function&& function) override
+    {
+        on_receive_cb_fn_ = std::move(function);
+    }
```

Comment:
This new  "callback" way API is needed for presentation layer by the `SubscriberImpl`.